src/main/java/frc/robot/subsystems/Swerve.java

```diff
@@ -0,0 +1,309 @@
+                            translationX.getAsDouble() * swerveDrive.getMaximumChassisVelocity(),
+                            translationY.getAsDouble() * swerveDrive.getMaximumChassisVelocity()), 0.8),
+                    Math.pow(angularRotationX.getAsDouble(), 3) * swerveDrive.getMaximumChassisAngularVelocity(),
+                    false,
+                    false);
+        },() -> {
+            for (SwerveModule module : swerveDrive.getModules()) {
+                module.getDriveMotor().set(0);
+                module.getAngleMotor().set(0);
+            }
+        });
+    }
+
+    public Pose2d getPose()
+    {
+        return swerveDrive.getPose();
+    }
+
+    public void resetOdometry(Pose2d initialHolonomicPose)
+    {
+        swerveDrive.resetOdometry(initialHolonomicPose);
+    }
+
+    public ChassisSpeeds getRobotVelocity()
+    {
+        return swerveDrive.getRobotVelocity();
+    }
+
+    public void resetOdometeryToStart() {
+        swerveDrive.swerveDrivePoseEstimator.resetPosition(
+                swerveDrive.getOdometryHeading(),
+                swerveDrive.getModulePositions(),
+                Pose2d.kZero
+        );
+    }
+
+    public void setUpPathPlanner(){
+        RobotConfig config;
+        try
+        {
+            config = RobotConfig.fromGUISettings();
+            AutoBuilder.configure(
+                    this::getPose,
+                    this::resetOdometry,
+                    this::getRobotVelocity,
+                    this::setSpeeds,
+                    new PPHolonomicDriveController(
+                            new PIDConstants(5.0, 0.0, 0.0),
+                            new PIDConstants(5.0, 0.0, 0.0)
+                    ),
+                    config,
+                    () -> {
+                        var alliance = DriverStation.getAlliance();
+                        if (alliance.isPresent())
+                        {
+                            return alliance.get() == DriverStation.Alliance.Red;
+                        }
+                        return false;
+                    },
+                    this
+            );
+        } catch (Exception e)
+        {
+            throw new Error(e);
+        }
+        PathfindingCommand.warmupCommand().schedule();
+    }
+
+    public ChassisSpeeds getSpeeds(){
+        return swerveDrive.getRobotVelocity();
+    }
+
+    public Command centerModules() {
+        return run(() -> Arrays.asList(swerveDrive.getModules())
+                .forEach(it -> it.setAngle(0.0)));
+    }
+
+
+    @Override
+    public void periodic() {
+        SwerveModulePosition[] modulePositions = swerveDrive.getModulePositions();
+        swerveDrive.updateOdometry();
+
+        for (int i = 0; i < modulePositions.length; i++) {
+            moduleMechanisms[i].setAngle(modulePositions[i].angle.getDegrees() + 90);
+        }
+
+    }
+    public void updatePoseEstimator(Optional<LimelightHelpers.PoseEstimate> optionalPose){
```

Comment:
this is a weird way to use `Optional`. This method should just receive `LimelightHelpers.PoseEstimate`... 

handle the optional in the caller instead.